rust/agama-lib/share/profile.schema.json

```diff
@@ -568,26 +560,227 @@
+        },
+        {
+          "title": "Size unit.",
+          "$ref": "#/$defs/sizeValue"
+        },
+        {
+          "title": "Size range.",
+          "description": "Lower size limit and optionally upper size limit.",
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/sizeValue"
+          },
+          "minItems": 1,
+          "maxItems": 2,
+          "examples": [[1024, 2048], ["1 GiB", "5 GiB"], [1024, "2 GiB"], ["2 GiB"]]
+        },
+        {
+          "title": "Size range.",
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["min"],
+          "properties": {
+            "min": {
+              "title": "Mandatory lower size limit.",
+              "$ref": "#/$defs/sizeValue"
+            },
+            "max": {
+              "title": "Optional upper size limit.",
+              "$ref": "#/$defs/sizeValue"
+            }
+          }
+        }
+      ]
+    },
+    "search": {
+      "title": "Search options",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name"],
+      "properties": {
+        "name": {
+          "title": "Device name",
+          "type": "string",
+          "examples": ["/dev/vda"]
+        }
+      }
+    },
+    "boot": {
+      "title": "Boot options.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["configure"],
+      "properties": {
+        "configure": {
+          "title": "Whether to configure partitions for booting.",
+          "type": "boolean"
+        },
+        "device": {
+          "title": "Device to use for booting.",
+          "description": "The installation device is used by default for booting.",
+          "type": "string",
+          "examples": ["/dev/vda"]
+        }
+      }
+    },
+    "encryptionMethod": {
+      "title": "Method used to create the encryption device.",
+      "enum": ["luks2", "tpm_fde"]
+    },
+    "encryptionPbkdFunction": {
+      "title": "Password-based key derivation function to use for LUKS2.",
+      "enum": ["pbkdf2", "argon2i", "argon2id"]
+    },
+    "encryption": {
+      "title": "Encryption options.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["key"],
+      "properties": {
+        "key": {
+          "title": "Passphrase to use when creating a new encryption device.",
+          "type": "string"
+        },
+        "method": {
+          "$ref": "#/$defs/encryptionMethod"
+        },
+        "pbkdFunction": {
+          "$ref": "#/$defs/encryptionPbkdFunction"
+        },
+        "label": {
+          "title": "LUKS label for the encrypted device.",
+          "type": "string"
+        },
+        "cipher": {
+          "title": "Cipher used for LUKS encryption.",
+          "description": "The value must be compatible with the --cipher argument of the command cryptsetup.",
+          "type": "string"
+        },
+        "key_size": {
```

Comment:
Is snake_case expected here? I thought we were using camelCase in all the schema (like pbkdFunction just some lines above).

Reply:
Good point.